glusterd2/xlator/options/options.go

```diff
@@ -324,10 +324,84 @@ func ValidateStr(o *Option, val string) error {
 
 // ValidateTime validates if the option is valid time format
 func ValidateTime(o *Option, val string) error {
-	if validate.IsTime(val, "hh:mm:ss") != true {
-		return ErrInvalidArg
+	var time int
+	var t string
+	var err error
+	if strings.HasSuffix(val, "sec") {
+		value := strings.TrimSuffix(val, "sec")
+		err = ValidateRange(o, value)
+		if err != nil {
+			return err
+		}
+	} else if strings.HasSuffix(val, "s") {
+		value := strings.TrimSuffix(val, "s")
+		err = ValidateRange(o, value)
+		if err != nil {
+			return err
+		}
+	} else if strings.HasSuffix(val, "min") {
+		t = strings.TrimSuffix(val, "min")
+		time, err = strconv.Atoi(t)
+		if err != nil {
+			return err
+		}
+		time = time * 60
+	} else if strings.HasSuffix(val, "m") {
+		t = strings.TrimSuffix(val, "m")
+		time, err = strconv.Atoi(t)
+		if err != nil {
+			return err
+		}
+		time = time * 60
+	} else if strings.HasSuffix(val, "hr") {
+		t = strings.TrimSuffix(val, "hr")
+		time, err = strconv.Atoi(t)
+		if err != nil {
+			return err
+		}
+		time = time * 60 * 60
+	} else if strings.HasSuffix(val, "h") {
+		t = strings.TrimSuffix(val, "h")
+		time, err = strconv.Atoi(t)
+		if err != nil {
+			return err
+		}
+		time = time * 60 * 60 * 24
+	} else if strings.HasSuffix(val, "days") {
+		t = strings.TrimSuffix(val, "days")
+		time, err = strconv.Atoi(t)
+		if err != nil {
+			return err
+		}
+		time = time * 60 * 60 * 24
+	} else if strings.HasSuffix(val, "d") {
+		t = strings.TrimSuffix(val, "d")
+		time, err = strconv.Atoi(t)
+		if err != nil {
+			return err
+		}
+		time = time * 60 * 60 * 24
+	} else if strings.HasSuffix(val, "w") {
+		t = strings.TrimSuffix(val, "w")
+		time, err = strconv.Atoi(t)
+		if err != nil {
+			return err
+		}
+		time = time * 60 * 60 * 24 * 7
+	} else if strings.HasSuffix(val, "wk") {
+		t = strings.TrimSuffix(val, "wk")
+		time, err = strconv.Atoi(t)
+		if err != nil {
+			return err
+		}
+		time = time * 60 * 60 * 24 * 7
+	} else {
```

Comment:
If there's no valid suffix, is it an error ?

Reply:
no. If the sufix is not given we are supposed to default it to seconds.

Reply:
The `else` case applies to both: 
* suffix not given and
* ANY other suffix like say `y`

But that's okay as `ValidateRange` shall fail.

Reply:
okay